provider/openstack/provider.go

```diff
@@ -562,6 +563,7 @@ func (e *Environ) ConstraintsValidator(ctx context.ProviderCallContext) (constra
 	for i, flavor := range flavors {
 		instTypeNames[i] = flavor.Name
 	}
+	sort.Strings(instTypeNames)
```

Comment:
I would just use the same sort as for ec2. Without delimiters it should be the same anyway.
Might as well add it to the other providers too.

Reply:
As they are not using the "instanceTypes" types I would need to add this to the underlying one of "flavor". Is that what you expect?

Reply:
Missed that. 
We *could* convert the flavours to `InstanceTypes`, but I think it's fine to leave it as it is.